packages/hebao_v1/contracts/base/MetaTxModule.sol

```diff
@@ -101,27 +101,39 @@ contract MetaTxModule is BaseModule
         uint startGas = gasleft();
         require(startGas >= gasSetting[2], "OUT_OF_GAS");
 
-        address wallet = extractWalletAddress(data);
+        // Copy data to memory
+        bytes memory _data = data;
+
+        address wallet = extractWalletAddress(_data);
         bytes32 metaTxHash = getSignHash(
             wallet, // from
             address(this),  // to. Note the relayer can only call its own methods.
             msg.value, // value
-            data,
+            _data,
             nonce,
             gasSetting
         );
 
-        address[] memory signers = extractMetaTxSigners(wallet, extractMethod(data), data);
-        metaTxHash.verifySignatures(signers, signatures);
+        bytes4 method = extractMethod(_data);
+        // If we exectute multiple transactions using `executeTransactions` we don't need to check
+        // any signatures now, we will check them for each transaction independently
+        if (method != this.executeTransactions.selector) {
+            address[] memory signers = extractMetaTxSigners(wallet, method, _data);
+            metaTxHash.verifySignatures(signers, signatures);
+        } else {
+            // As `metaTxHash` is the hash of `data` it can't contain the hash of `data`, so set it here
+            require(metaTxHash == 0x0, "INVALID_DATA");
```

Comment:
metaTxHash  is calculated by getSignHash, so it's unlikely it will be 0x0.

Reply:
It needs to be temporarily set to `0x0` because `data` is used to calculate `metaTxHash`. So `metaTxHash` can't be part of `data` otherwise we modify the data that is used to calculate the hash, which isn't possible of course. :)

So my solution to this was to set it to `0x0` for the hash calculation and then set it to the correct value on-chain. I would also like to find another way to do this.

Reply:
is it possible to do it this way. 

```
if (method != this.executeTransactions.selector) {
   metaTxHash = getSignHash()...
} else {
  //do nothing
}

```
then in executeTransactions, drop the metaTxHash parameter, and calculate the hash inside the function using it's own selector and parameters...

Reply:
I have found a way to make it completely general without any changes to `executeMetaTx` which I think is very nice because `executeTransactions` is now a module function like any other. If a module doesn't want to support or a different function for multiple calls is wanted for some reason that's very easy now.